lib/index.js

```diff
@@ -723,16 +746,38 @@ function _connect(options, callback) {
     };
   }
 
-  mongo.MongoClient.connect(options.url, connectOptions, (err, db) => {
+  MongoClient.connect(options.url, connectOptions, (err, client) => {
     if(!err && !options.init) {
       logger.info('connecting to database: ' + options.url);
     }
-    callback(err, db);
+    const db = client.db(config.name);
+    callback(err, {db, client});
+  });
+}
+
+/**
+ * Logs in a user using an auth object.
+ *
+ * @param {object} options - Options to use.
+ * @param {object} options.auth - user and password credentials.
+ * @param {object} options.opts - Options for the MongoClient.
+ * @param {Function} options.callback - A callback function.
+ * @param {Server} options.server - The mongo server to connect to.
+ *
+ * @returns {Promise} The result of the connect.
+*/
+function _loginUser({auth, opts, callback, server}) {
```

Comment:
Since this returns a promise, we should mark it `async`, it helps with stack traces.

```suggestion
async function _loginUser({auth, opts, callback, server}) {
```

Reply:
I really can't believe I missed that one, but I made the correct here: https://github.com/digitalbazaar/bedrock-mongodb/pull/33/commits/4733ee5f8dc3799abfe26b0bb721325cdea81a1d I guess having the callback confused me?